Add docs for local simulation, benchmarking and gas profiling

vgao1996 opened the pull request:
This adds the proper documentation for the new local simulation features added in https://github.com/aptos-labs/aptos-core/pull/12832.

hariria commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
```diff
@@ -0,0 +1,251 @@
+    <FileTree.File name="Move.toml" />
+    <FileTree.Folder name="sources" />
+    <FileTree.Folder name="gas-profiling" defaultOpen>
+      <FileTree.Folder name="txn-XXXXXXXX-0xXXXX-message-set_message"
+                       defaultOpen>
```
Nit: is this supposed to be on two lines?

hariria reviewed (approved):
This looks fantastic! Thanks for adding all the filenames to codeblocks and being meticulous! Some nits

georgemitenkov commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
```diff
@@ -0,0 +1,251 @@
+- `aptos move publish`
+
+<Callout type="info" emoji="ℹ️">
+Local simulations do not result in any changes of the chain state.
```
nit: `to the on-chain state`?

georgemitenkov commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
```diff
@@ -0,0 +1,251 @@
+</Callout>
+
+## Deploying the Example Contract
+For demonstration purposes, we will continue to use the [`hello_blockchain`](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/hello_blockchain) package as an example.
```
This doc talks about local simulation, benchmarking, etc. should we mention that it can be done without deployment to testnet/devnet (can it)?

vgao1996 replied:
Added a note.

georgemitenkov commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
```diff
@@ -0,0 +1,251 @@
+The previous tutorial demonstrates how you can deploy and interact with Move contracts using various CLI commands.
+
+By default, those commands send a transaction to the remote fullnode for simulation and execution.
+You can override this behavior and simulate the transaction locally, by appending the following command line options of your preference:
```
nit: by appending  *one of the* command line options? they are all local and mutually exclusive, right?

georgemitenkov commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
````diff
@@ -0,0 +1,251 @@
+To measure the running time(s) of your transaction, use the `--benchmark` option.
+
+```bash filename="Terminal"
+aptos move run --function-id 'default::message::set_message' --args 'string:abc' --benchmark
````
It would be nice to mention that we can replay and benchmark / profile gas any transaction at version X - I think it is a great feature. Not sure if in this doc because it talks about local simulations, but we should mention that 100%

vgao1996 replied:
There will be a dedicated PR for the new replay command. It will come in a subsequent PR!

georgemitenkov commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
```diff
@@ -0,0 +1,251 @@
+  </FileTree.Folder>
+</FileTree>
+
+`index.html`` is the main page of the report, which can view using your web browser.
```
typo? you have `` here after `index.html`

georgemitenkov commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
```diff
@@ -0,0 +1,251 @@
+
+### Understanding the Gas Report
+
+The gas report consists of three parts, enabling you to understand the gas usage through different lenses.
```
nit: three parts and helps you ....

georgemitenkov replied:
also sections, not parts, to be consistent with the naming below

georgemitenkov commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx:
```diff
@@ -0,0 +1,251 @@
+
+The middle column represents the gas costs associated with the operations.
+
+There is also a special notation `@number` that represents a jump to a particular location in the byte code.
```
curious, are there also line numbers in this table? to understand where `@number` jumps to?

vgao1996 replied:
The number here is the offset to a bytecode location, so it can't be easily shown here. In the future we should just go a step further and show people annotated source code.

georgemitenkov reviewed (approved):
Thanks, looks good! Found a few typos and left some comments/suggestions

hariria commented:
@vgao1996 is this ready to be merged?

vgao1996 commented:
@hariria need to make some final adjustments to address the issues @georgemitenkov brought up. Please allow me a bit more time on this

vgao1996 commented:
@hariria @georgemitenkov I've addressed all the issues you brought up. Thank you both for the great feedback!

Will merge the PR soon once all tests pass. Let me know if you intend to request last minute changes
